kedro/templates/project/hooks/utils.py

```diff
@@ -141,13 +144,36 @@ def _remove_pyspark_viz_starter_files(is_viz: bool, python_package_name: str) ->
     _remove_file(test_pipeline_path)
 
 
-def setup_template_tools(selected_tools_list: str, requirements_file_path: str, pyproject_file_path: str, python_package_name: str, example_pipeline: str) -> None:
-    """Setup the templates according to the choice of tools.
+def _remove_extras_from_kedro_datasets(file_path: Path) -> None:
+    """Remove all extras from kedro-datasets in the requirements file.
+
+    Args:
+        file_path (Path): The path of the requirements file.
+    """
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    new_lines = []
```

Comment:
Thanks for PR, @SajidAlamQB ! I have a small question, is lines a List? If yes why you are creating a new List new_lines and copying, not just modifying 1 row in list variable?

Reply:
It was for readability but can amend for simplification.

Reply:
Simplified.